app/navigation/Navigator.scala

```diff
@@ -86,46 +88,61 @@ class Navigator @Inject() () {
       }
       .getOrElse(routes.JourneyRecoveryController.onPageLoad())
 
-  private def navigateFromAssessment(assessmentPage: AssessmentPage)(answers: UserAnswers): Call = {
+  private def navigateFromHasSupplementaryUnit(recordId: String)(answers: UserAnswers): Call =
+    answers
+      .get(HasSupplementaryUnitPage(recordId))
+      .map {
+        case true  => routes.SupplementaryUnitController.onPageLoad(NormalMode, recordId)
+        case false => routes.CyaCategorisationController.onPageLoad(recordId)
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
+
+  private def navigateFromAssessment(assessmentPage: AssessmentPage, mode: Mode)(answers: UserAnswers): Call = {
+    val recordId = assessmentPage.recordId
+
     for {
       recordQuery      <- answers.get(RecordCategorisationsQuery)
       assessmentAnswer <- answers.get(assessmentPage)
     } yield assessmentAnswer match {
       case AssessmentAnswer.Exemption(_) =>
         val assessmentCount = Try {
-          recordQuery.records
-            .get(assessmentPage.recordId)
-            .get
+          recordQuery.records(recordId)
             .categoryAssessments
             .size
         }.getOrElse(0)
+
         if (assessmentPage.index + 1 < assessmentCount) {
-          routes.AssessmentController.onPageLoad(NormalMode, assessmentPage.recordId, assessmentPage.index + 1)
+          if (mode == CheckMode && answers.isDefined(AssessmentPage(recordId, assessmentPage.index + 1))) {
```

Comment:
You should separate out checkmode and normalmode out into separate functions like they have done for all the other functions in navigation

Reply:
I did at first but I kinda hate redoing all the assessmentPage logic in both places. Happy to if that's the pattern we want to go with though

Reply:
Done